pkg/utils/version/version.go

```diff
@@ -57,3 +57,34 @@ func CheckVersionMeetsConstraint(version, constraint string) (bool, error) {
 
 	return c.Check(v), nil
 }
+
+// Range represents a version range of type [MinVersion, MaxVersion).
+type Range struct {
+	MinVersion string
+	MaxVersion string
+}
+
+// Contains returns true if the range contains the given version, false otherwise.
+// The range contains the given version only if it's greater or equal than MinVersion (always true if MinVersion is empty),
+// and less than MaxVersion (always true if MaxVersion is empty).
+func (r *Range) Contains(version string) (bool, error) {
```

Comment:
It seems you could replace most of this function with a direct usage of `CheckVersionMeetsConstraint` instead of `CompareVersions`. A constraint string may consist of multiple assertions, allowing for a direct range check: https://github.com/Masterminds/semver/blob/master/constraints_test.go#L321

Reply:
Good idea, I rewrote it this way, although because I want to treat empty `MinVersion` and `MaxVersion` in a special way I need some logic to calculate the constraint, so it's a bit more concise this way, but not by much.